File: BaseLibrary/equip/Armor.cs
namespace BaseLibrary.equip
{
    public class Armor : IEquipment{
        /// <summary>
        /// How taught is armor. How many hit points can robot have.
        /// </summary>
        public int MAX_HP { get; private set; }

        /// <inheritdoc />
        public int COST { get; private set; }

        /// <inheritdoc />
        public int ID { get; private set; }

        public Armor(int ID, int COST, int MAX_HP) {
            this.MAX_HP = MAX_HP;
            this.COST = COST;
            this.ID = ID;
        }

        protected bool Equals(Armor other) {
            return MAX_HP == other.MAX_HP && COST == other.COST && ID == other.ID;
        }

        public override bool Equals(object obj) {
            if (ReferenceEquals(null, obj)) return false;
            if (ReferenceEquals(this, obj)) return true;
            if (obj.GetType() != this.GetType()) return false;
            return Equals((Armor) obj);
        }

        public override int GetHashCode() {
            unchecked {
                int hashCode = MAX_HP;
                hashCode = (hashCode * 397) ^ COST;
                hashCode = (hashCode * 397) ^ ID;
                return hashCode;
            }
        }

        public override string ToString() {
            return $"{nameof(MAX_HP)}: {MAX_HP}, {nameof(COST)}: {COST}, {nameof(ID)}: {ID}";
        }
    }
}

File: Java-Client/RobotBattlefield-Client-Java/src/BaseLibrary/equipment/Armor.java
package BaseLibrary.equipment;

import BaseLibrary.communication.protocol.AProtocol;

public class Armor implements IEquipment {
	
	private static final String NAME = "ARMOR";

	static {
		AProtocol.registerForDeserialize(NAME, Armor.class);
	}

	@Override
	public String getCommandName() {
		return NAME;
	}
	
	/**
	 * How taught is armor. How many hit points can robot have.
	 */
	public final int MAX_HP;

	public final int COST;

	public final int ID;


	public Armor(int ID, int COST, int MAX_HP) {
		this.MAX_HP = MAX_HP;
		this.COST = COST;
		this.ID = ID;
	}

	protected boolean equals(Armor other) {
		if (COST != other.COST)
			return false;
		if (ID != other.ID)
			return false;
		if (MAX_HP != other.MAX_HP)
			return false;
		return true;
	}

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + COST;
		result = prime * result + ID;
		result = prime * result + MAX_HP;
		return result;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Armor other = (Armor) obj;
		return equals(other);
	}
	@Override
	public int getCost() {
		return COST;
	}
	@Override
	public int getID() {
		return ID;
	}

	/**
	 * How taught is armor. How many hit points can robot have.
	 */
	public int getMaxHP() {
		return MAX_HP;
	}
}
